src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,22 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import mate.academy.hibernate.relations.util.HibernateUtil;
 
 public class CountryServiceImpl implements CountryService {
+    private final CountryDao countryDao = new CountryDaoImpl(HibernateUtil.getSessionFactory());
+
     @Override
     public Country add(Country country) {
-        return null;
+        countryDao.add(country);
+        return country;
```

Comment:
According to the checklist, you should add a `catch` block for operations on the DAO layer. Wrap the `countryDao.add(country)` call in a try-catch block to handle any possible exceptions.